steward/src/cork/proposals.rs

```diff
@@ -81,292 +101,220 @@ impl ProposalThreadState {
+                        );
+                        state.increment_proposal_id();
                         continue;
                     }
-                };
-                if governance_call.call.is_none() {
-                    error!("governance call is empty: {:?}", governance_call);
-                    continue;
                 }
-                let encoded_call: Vec<u8> = match governance_call.call.unwrap() {
-                    Call::AaveV2Stablecoin(data) => {
-                        if data.function.is_none() {
-                            error!(
-                                // if this happens there is a problem with proposal validation in sommelier
-                                "cork proposal {} call data contains no function data: {:?}",
-                                proposal.proposal_id, data,
-                            );
-                            continue;
-                        }
-                        let function = data.function.unwrap();
-                        aave_v2_stablecoin::get_encoded_governance_call(
-                            function,
-                            &cellar_id,
-                            proposal.proposal_id,
-                        )?
-                    }
-                    Call::CellarV1(data) => {
-                        if data.function.is_none() {
-                            error!(
-                                "cork proposal {} call data contains no function data: {:?}",
-                                proposal.proposal_id, data
-                            );
-                            continue;
-                        }
-                        let function = data.function.unwrap();
-                        cellar_v1::get_encoded_governance_call(
-                            function,
-                            &cellar_id,
-                            proposal.proposal_id,
-                        )?
-                    }
-                };
-                debug!(
-                    "proposal {} encoded call: {:?}",
-                    proposal.proposal_id, encoded_call
-                );
+            }
+        };
 
-                if let Err(err) =
-                    schedule_cork(&cellar_id, encoded_call.clone(), block_height).await
-                {
-                    error!(
-                        "failed to schedule cork for proposal {}, will retry shortly. reason: {}",
-                        proposal.proposal_id, err
-                    );
-                    // don't add an item twice to retry data
-                    for d in state.retry_data.iter() {
-                        if d.0 == proposal.proposal_id {
-                            continue;
-                        }
+        debug!(
+            "proposal {} encoded call: {:?}",
+            proposal.proposal_id, encoded_call
+        );
+
+        if let Err(err) = schedule_cork(&cellar_id, encoded_call.clone(), block_height).await {
```

Comment:
I opted to let the polling interval be a "built in" retry, had actually forgot about the config value. I'll add a loop.

Reply:
Yeah, the thing is that the polling interval as retry doesn't have a stopping point. If you want it to move on, it would be preferable for it to do so after a limited period of time or attempts rather than waiting until the block height it was supposed to execute at before looking past that proposal ID.